sdk/storage/azure-storage-blob/src/main/java/com/azure/android/storage/blob/transfer/BlockDownloadEntity.java

```diff
@@ -0,0 +1,214 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.android.storage.blob.transfer;
+
+import androidx.room.ColumnInfo;
+import androidx.room.Entity;
+import androidx.room.ForeignKey;
+import androidx.room.Ignore;
+import androidx.room.Index;
+import androidx.room.PrimaryKey;
+import androidx.room.TypeConverters;
+
+import com.azure.android.core.util.Base64Util;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.UUID;
+
+import static androidx.room.ForeignKey.CASCADE;
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+/**
+ * Package private.
+ *
+ * Represents metadata for a block download.
+ *
+ * There is one-to-many relationship between a {@link BlobDownloadEntity} and a set of {@link BlockDownloadEntity}.
+ * Each {@link BlockDownloadEntity} in the set describes a chunk of the file referenced by the
+ * {@link BlobDownloadEntity}
+ *
+ * The Data Access Object type {@link DownloadDao} exposes DB store and read methods on this model.
+ *
+ * @see TransferDatabase
+ */
+@Entity(tableName = "blockdownloads",
+    foreignKeys = @ForeignKey(entity = BlobDownloadEntity.class,
+        parentColumns = "key",
+        childColumns = "blob_key",
+        onDelete = CASCADE),
+    indices = {@Index("blob_key")})
+final class BlockDownloadEntity {
+    /**
+     * A unique key for the block download metadata.
+     *
+     * This key identifies the metadata in the local store, which is different from
+     * {@link BlockDownloadEntity#blockId}. Block ID is used by the storage service to uniquely identify the block.
```

Comment:
this comment may not be applicable for download since we never have to send it to service.